extensions/vscode/src/events.ts

```diff
@@ -231,16 +231,19 @@ export class EventStream extends Readable implements Disposable {
           newMessage.type = "publish/restoreEnv/progress";
           break;
         default:
-          throw new Error(
-            `events::messageFactory: Unknown publish/restoreREnv based message: ${newMessage.type}`,
+          newMessage.type = "undefined";
```

Comment:
This feels very strange to me.

Reply:
>I don't personally like returning undefined if there is an error.

We're not really returning undefined, it is just the fact that we can't determine what the type is for these restore messages would be. So the type of the message gets set to undefined, which seems right, but the message continues to be processed. An alternative to this approach could be that we swallow the unknown message, but I thought this was better.